tiledb/sm/subarray/tile_cell_slab_iter.cc

```diff
@@ -267,7 +267,6 @@ void TileCellSlabIter<T>::init_cell_slab_lengths() {
           ranges_[dim_num_ - 1][i].end_ - ranges_[dim_num_ - 1][i].start_ + 1;
     }
   } else {
-    assert(layout_ == Layout::COL_MAJOR);
```

Comment:
I am skeptical of this, for two reasons:

First, `assert` is almost always optimized out in release builds. It uses a C preprocessor directive to expand to nothing. We have observed the failure in release builds, which means either this assert is not at fault, or we are not abiding that pattern.

Second, if the `assert` did fail, then we would see more catastrophic reporting of a `SIGABRT` from Catch2 or CI.  Either that or we need to find a new testing library.

Does this assert fail deterministically in the test? Or is there a memory corruption which causes it to occasionally fail?

Reply:
As stated in the PR description, this change was specifically caught in Debug mode; I hit it every time I ran the test. I was unable to reproduce any other failure in Release mode in 20,000 iterations of the test across 4 different builds.

Reply:
This sounds like three separate problems to me.
1. The intermittent failure, which we observe in release mode; the assert does not run here and removing it will not fix the intermittent failure.
2. The correctness of this assert, which we see failing deterministically in debug mode; I would like to see more description here about what the assert claims, why it is failing and whether removing it is correct.
3. The fact that this fails deterministically in debug mode; we should ensure somehow that our asserts run in CI rather than in development only.  I don't think this is an action item for this PR but does require someone to follow up.

Reply:
`TILEDB_ASSERTIONS` are default-disabled. AFAICT, no failing builds have `--enable-assertions` in bootstrap. This was intended to be a separate "fix" of something discovered in `Debug`, but I think you may be right that it's worth deeper discovery.

Reply:
I discovered that recently as well, in relation to [this discussion](https://tiledb.slack.com/archives/C7STPC519/p1747256956794759)
sorry I forgot to circle back here and mention it :/